go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -294,17 +309,24 @@ func localnetNodePortWatcher(ipt util.IPTablesHelper, wf *factory.WatchFactory)
 	return err
 }
 
-func cleanupLocalnetGateway() error {
+func cleanupLocalnetGateway(physnet string) error {
 	// get bridgeName from ovn-bridge-mappings.
 	stdout, stderr, err := util.RunOVSVsctl("--if-exists", "get", "Open_vSwitch", ".",
 		"external_ids:ovn-bridge-mappings")
 	if err != nil {
 		return fmt.Errorf("Failed to get ovn-bridge-mappings stderr:%s (%v)", stderr, err)
 	}
-	bridgeName := strings.Split(stdout, ":")[1]
-	_, stderr, err = util.RunOVSVsctl("--", "--if-exists", "del-br", bridgeName)
-	if err != nil {
-		return fmt.Errorf("Failed to ovs-vsctl del-br %s stderr:%s (%v)", bridgeName, stderr, err)
+	bridge_mappings := strings.Split(stdout, ",")
+	for _, bridge_mapping := range bridge_mappings {
+		m := strings.Split(bridge_mapping, ":")
+		if physnet == m[0] {
+			bridgeName := m[1]
+			_, stderr, err = util.RunOVSVsctl("--", "--if-exists", "del-br", bridgeName)
+			if err != nil {
+				return fmt.Errorf("Failed to ovs-vsctl del-br %s stderr:%s (%v)", bridgeName, stderr, err)
+			}
+		}
```

Comment:
You can return at L328 with `return nil` since there can't be two entries for physicalNetworkName